main.tf

```diff
@@ -0,0 +1,104 @@
+data "aws_caller_identity" "current" {
+}
+
+locals {
+  account_id = data.aws_caller_identity.current.account_id
+}
+
+data "aws_ami" "byol" {
+  most_recent = true
+  owners      = ["679593333241"]
+
+  filter {
+    name   = "name"
+    values = ["FortiGate-VM64-AWS build*"]
+  }
+}
+
+data "aws_ami" "ondemand" {
+  most_recent = true
+  owners      = ["679593333241"]
+
+  filter {
+    name   = "name"
+    values = ["FortiGate-VM64-AWSONDEMAND*"]
+  }
+}
+
+data "aws_region" "current" {}
+
+data "template_file" "userdata" {
+  template = file("${path.module}/userdata.tpl")
+  vars = {
+    bucket      = var.config_bucket_name
+    region      = var.config_bucket_region == "" ? data.aws_region.current.name : var.config_bucket_region
+    license     = var.use_byol ? var.config_bucket_license_file : ""
+    config_file = var.config_bucket_config_file
+  }
+}
+
+locals {
+  latest_ami = var.use_byol ? data.aws_ami.byol.id : data.aws_ami.ondemand.id
+  ami        = coalesce(var.override_ami, local.latest_ami)
+}
+
+module "fortigate_password" {
+  source           = "git::https://github.com/rhythmictech/terraform-aws-secretsmanager-random-secret?ref=v1.1.1"
+  create_secret    = var.load_default_config
+  name_prefix      = var.name
+  description      = "${var.name} admin password"
+  length           = 20
+  override_special = "@#$%^*()-=_+[]{};<>?,./"
+  tags             = var.tags
+}
+
+module "keypair" {
+  source      = "git::https://github.com/rhythmictech/terraform-aws-secretsmanager-keypair?ref=v0.0.3"
+  name_prefix = var.name
+  description = "${var.name} SSH keypair"
+  tags        = var.tags
+}
+
+resource "aws_instance" "this" {
+  ami                  = local.latest_ami
+  ebs_optimized        = true
+  iam_instance_profile = aws_iam_instance_profile.this.name
+  instance_type        = var.instance_type
+  key_name             = module.keypair.key_name
+  subnet_id            = var.internal_subnet_id
+  tags = merge(
+    var.tags, {
+      "Name" = var.name
+  })
+  user_data              = var.enable_auto_config ? data.template_file.userdata.rendered : ""
+  vpc_security_group_ids = [aws_security_group.internal.id]
+
+  depends_on = [aws_s3_bucket_object.default_config]
+
+  lifecycle {
+    prevent_destroy = true
```

Comment:
That's going to confuse somebody someday

Reply:
Actually it shouldn't, as if you just remove the module entirely, tf will still remove it. This mostly blocks you deleting your firewall config accidentally. A `terraform destroy` would sure fail though.